client/src/main/scala/org/apache/celeborn/client/commit/ReducePartitionCommitHandler.scala

```diff
@@ -93,7 +95,11 @@ class ReducePartitionCommitHandler(
   }
 
   override def setStageEnd(shuffleId: Int): Unit = {
-    stageEndShuffleSet.add(shuffleId)
+    getReducerFileGroupRequest synchronized {
+      stageEndShuffleSet.add(shuffleId)
+      getReducerFileGroupRequest.remove(shuffleId)
+        .asScala.foreach(replyGetReducerFileGroup(_, shuffleId))
```

Comment:
IMO we should move the invocation of replyGetReducerFileGroup out the synchronize block to avoid the critical section cost too much time.

Reply:
Updated